sdk/pinocchio/src/sysvars/mod.rs

```diff
@@ -44,3 +44,48 @@ macro_rules! impl_sysvar_get {
         }
     };
 }
+
+/// Handler for retrieving a slice of sysvar data from the `sol_get_sysvar`
+/// syscall.
+///
+/// # Safety
+///
+/// `dst` must point to a buffer that can hold at least `length` bytes.
+#[inline]
+pub unsafe fn get_sysvar_unchecked(
+    dst: *mut u8,
+    sysvar_id: &Pubkey,
+    offset: u64,
+    length: u64,
+) -> Result<(), ProgramError> {
+    let sysvar_id = sysvar_id as *const _ as *const u8;
+
+    #[cfg(target_os = "solana")]
+    let result = unsafe { crate::syscalls::sol_get_sysvar(sysvar_id, dst, offset, length) };
+
+    #[cfg(not(target_os = "solana"))]
+    let result = core::hint::black_box(sysvar_id as u64 + dst as u64 + offset + length);
```

Comment:
As this is a function call, and not a macro, please import all the dependencies at the top of the module, and use reasonably short names in the function body:

```rust
use core::hint::black_box;

use crate::{
    /* ... */,
    crate::syscalls::sol_get_sysvar,
    /* ... */,
};
```

```suggestion
    #[cfg(target_os = "solana")]
    let result = unsafe { sol_get_sysvar(sysvar_id, dst, offset, length) };

    #[cfg(not(target_os = "solana"))]
    let result = black_box(sysvar_id as u64 + dst as u64 + offset + length);
```

Reply:
I personally do not like it, as it breaks consistency.
There are now two ways to call any function: either directly by name, or via an absolute path.
Plus, your import block at the top is incomplete now as well.
I see it as an unnecessary complication.

In the other PRs that I review, I ask for this style to be avoided.
But it is the case that the code in the solana/agave repo is not super consistent.

So, I would say that the other locations should be changed to avoid absolute references in function calls as well.
But it is quite possible that other reviewers would approve this PR as is.